amworkflow/occ_helpers.py

```diff
@@ -191,3 +206,245 @@ def topo_explorer(shape: TopoDS_Shape, shape_type: str) -> list:
+        if (ymin + ymax < 1e-3) or (
+            abs((ymin + ymax) * 0.5 - bbox[4]) < 1e-3
+        ):  # if the edge is either
+            xx.append(e)
+        if (xmin + xmax < 1e-3) or (abs((xmin + xmax) * 0.5 - bbox[3]) < 1e-3):
+            yy.append(e)
+    edges = xx + yy
+    # build a compound of all edges
+    wire = create_compound(edges)
+    return wire
+
+def get_face_center_of_mass(face: TopoDS_Face, gp_pnt: bool = False) -> tuple:
+    """
+    @brief Get the center of mass of a TopoDS_Face. This is useful for determining the center of mass of a face or to get the centre of mass of an object's surface.
+    @param face TopoDS_Face to get the center of mass of
+    @param gp_pnt If True return an gp_Pnt object otherwise a tuple of coordinates.
+    """
+    props = GProp_GProps()
+    brepgprop_SurfaceProperties(face, props)
+    face_surf = props.CentreOfMass()
+    # face_surf. Coord if gp_pnt returns the face surface.
+    if gp_pnt:
+        return face_surf
+    else:
+        return face_surf.Coord()
+    
+def get_volume_center_of_mass(vol: TopoDS_Solid, gp_pnt: bool = False):
+    """
+    @brief Return the center of mass of a volume. This is an approximation of the centroid of the volume.
+    @param vol TopoDS_Solid object representing a volume
+    @param gp_pnt If True return an gp_Pnt object otherwise a tuple of coordinates.
+    @return Center of mass of a volume.
+    """
+    props = GProp_GProps()
+    brepgprop_VolumeProperties(vol, props)
+    cog = props.CentreOfMass()
+    # Return the current coordinate of the current coordinate system.
+    if gp_pnt:
+        return cog
+    else:
+        return cog.Coord()
+    
+def get_face_area(face: TopoDS_Face) -> float:
+    """
+    @brief Get the area of a TopoDS_Face. This is an approximation of the area of the face.
+    @param face to get the area of.
+    @return The area of the face.
+    """
+    props = GProp_GProps()
+    brepgprop_SurfaceProperties(face, props)
+    face_area = props.Mass()
+    return face_area
+
+def get_faces(_shape):
+    """
+    @brief Get faces of a shape. This is a list of topods_Face objects that correspond to the faces of the shape
+    @param _shape shape to get faces of
+    @return list of topods_Face objects ( one for each face in the shape ) for each face
+    """
+    topExp = TopExp_Explorer()
+    topExp.Init(_shape, TopAbs_FACE)
+    _faces = []
+
+    # Add faces to the faces list
+    while topExp.More():
+        fc = topods_Face(topExp.Current())
+        _faces.append(fc)
+        topExp.Next()
+
+    return _faces
+
+def get_point_coord(p: gp_Pnt) -> tuple:
+    """
+    @brief Returns the coord of a point. This is useful for debugging and to get the coordinates of an object that is a part of a geometry.
+    @param p gp_Pnt to get the coord of
+    @return tuple of the coordinate of the point ( x y z ) or None if not a point ( in which case the coordinates are None
+    """
+    return p.Coord()
```

Comment:
Why do you just use p.Coord() directly for debugging?

Reply:
Sorry that previously I wrote this functions just to help me remember that an OCC point object can be converted to a tuple by this approach. I would delete it now.